homeassistant/components/husqvarna_automower_ble/lawn_mower.py

```diff
@@ -0,0 +1,147 @@
+"""The Husqvarna Autoconnect Bluetooth lawn mower platform."""
+
+from __future__ import annotations
+
+from homeassistant.components import bluetooth
+from homeassistant.components.lawn_mower import (
+    LawnMowerActivity,
+    LawnMowerEntity,
+    LawnMowerEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, LOGGER
+from .coordinator import HusqvarnaCoordinator
+from .entity import HusqvarnaAutomowerBleEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up AutomowerLawnMower integration from a config entry."""
+    coordinator: HusqvarnaCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    address = coordinator.address
+
+    async_add_entities(
+        [
+            AutomowerLawnMower(
+                coordinator,
+                address,
+            ),
+        ]
+    )
+
+
+class AutomowerLawnMower(HusqvarnaAutomowerBleEntity, LawnMowerEntity):
+    """Husqvarna Automower."""
+
+    def __init__(
+        self,
+        coordinator: HusqvarnaCoordinator,
+        address: str,
+    ) -> None:
+        """Initialize the lawn mower."""
+        super().__init__(coordinator)
+        self._attr_name = None
+        self._attr_unique_id = str(address)
+        self._attr_supported_features = (
+            LawnMowerEntityFeature.PAUSE
+            | LawnMowerEntityFeature.START_MOWING
+            | LawnMowerEntityFeature.DOCK
+        )
+        self._attr_activity = LawnMowerActivity.ERROR
+
+    def _get_activity(self) -> LawnMowerActivity | None:
+        """Return the current lawn mower activity."""
+        if self.coordinator.data is None:
+            return None
+
+        state = str(self.coordinator.data["state"])
+        activity = str(self.coordinator.data["activity"])
```

Comment:
Would it be an idea to do this in the coordinator?

Reply:
I think you could use a dataclass to replace the `dict[str, bytes]` and make it more typesafe

Reply:
The Python library has been re-written to support [enums for these](https://github.com/alistair23/AutoMower-BLE/blob/main/automower_ble/protocol.py#L24)

Once this PR is merged the next step will be to update to the not yet released re-write. At that point this can be updated if that's ok. I'm trying to get this merged first before handling the API changes with a large library update

Reply:
Why don't we use the new lib already?

Reply:
Basically because this current PR is using the existing one and I want to get this merged :) I wanted to avoid changing this PR to try and keep things simple

Then there will be follow up work to update to the new lib and support more request types